internal/checker/checker.go

```diff
@@ -6413,7 +6418,7 @@ func (c *Checker) checkUnusedIdentifiers(potentiallyUnusedIdentifiers []*ast.Nod
 			}
 			c.checkUnusedTypeParameters(node)
 		case ast.KindMethodSignature, ast.KindCallSignature, ast.KindConstructSignature, ast.KindFunctionType, ast.KindConstructorType,
-			ast.KindTypeAliasDeclaration, ast.KindInterfaceDeclaration:
+			ast.KindTypeAliasDeclaration, ast.KindJSTypeAliasDeclaration, ast.KindInterfaceDeclaration:
```

Comment:
Every time we have a KindTypeAliasDeclaration, we should have a KindJSTypeAliasDeclaration? How does that scale for getting things right long-term? Hopefully it's not too much worse than the old code...

Reply:
Yes, except for the places where we shouldn't. I looked at the current codebase and I'm confident that it's 95% right and that the tests catch the rest.

The real value of having two kinds is in commonjs, where `module.exports=` is going to need to allow a bunch of things that `export=` doesn't. 
Still, it's ultimately a matter of whether jstype should work the same as type by default or not, not expressiveness, because you can always have `isJS := node.Flags & FlagsReparsed != 0`. That way jstype works like type by default.

Reply:
In other words, as an exercise for *me*, when developing, having two kinds is GREAT, because it forces me to go port every single TS construct to JS and consider the semantics. And if I forget something, the tests won't work.

However, long-term it might make more sense to have everything work the same by default.

Reply:
After discussing this in the design meeting: Anders prefers having two separate kinds and nobody else spoke up.